### ISSUE TITLE
Is it possible to change the encoding for saving gcode to UTF-8?

### ISSUE THREAD

**Issue by wtda** [sb53systems/G-Code-Flow-Temperature-Controller#11]
In Chinese system, it seems that the generated gcode is not saved in UTF-8 format, resulting in no data underneath, and klipper can't recognise it and print it (can't see the thumbnail). Manually change ‘Output.gcode’ to UTF-8 format before it can be recognised by fluidd (can see thumbnails) and printed.
Is it possible to change the saved encoding to UTF-8?
Translated by DeepL :)

**Comment by wtda** [comment]
update:
When I enable:“Beta: Use Unicode UTF-8 to get global language support”, it works!
Another question:
“UsedFilamentandExtruderintheG-Codefilearenot found ! choose your settings and click on the Refresh and generate button”
How do I include this data when slicing？

**Comment by sb53systems** [comment]
> update:
> When I enable:“Beta: Use Unicode UTF-8 to get global language support”, it works!
> Another question:
> “UsedFilamentandExtruderintheG-Codefilearenot found ! choose your settings and click on the Refresh and generate button”
> How do I include this data when slicing？

Which slicer are you using ?

**Comment by sb53systems** [comment]
Orca Slicer Include this information in the G-Code  
![image](https://github.com/user-attachments/assets/69146481-46cc-4337-ab27-49cd5d685a0a)  
![image](https://github.com/user-attachments/assets/c2fbea0f-ce40-4a8e-aa4b-2a9d9934d37a)  
  
in the script, you have to create the preset with the same name as in the Slicer  
![image](https://github.com/user-attachments/assets/0da6a9e0-9433-4eb5-a76f-ed5abb0861c7)

**Comment by sb53systems** [comment]
send me a g-code example file with characters you use.  
I would do my best to correct the problem

**Comment by wtda** [comment]
Thanks for the reply.
Comparing your screenshot with my own file, I found the problem: I have a TradRack multicolour system, so I have multiple filament name after ‘filament_settings_id’
When I use only one filament, it works fine~

**Comment by sb53systems** [comment]
@wtda , If you managed to get the script to work with your conditions, when you have time, could you summarize the steps to follow ? for someone who encountered the same problem.  
  
Since this script is only an example solution, and must be developed and integrated into the Slicers, I can't adapt it to all cases. 
But if you think any improvement is required at the earliest, let me know and I'll add it to the priority list. Thank you.  
  
Best regards.  
Salim.
